gc/2021/GC-06-29.md

```diff
@@ -0,0 +1,240 @@
+
+AR: acceptable to say if you have mutually recursive modules, you need to compile into single module, which is what some modules are doing
+
+TL: if we do that, and acceptable, nominal types will just work
+
+AR: no, that discussion is not dependent on nominal types
+
+CW: will remove mutual recursion as objection to nominal types
+
+TL: weak import export, refactoring of modules, for these solutions, you need to know transitive dependencies to break cycles at build types, if we say all mutually recursive modules need to be compiled into same module, then becomes non-issue.
+
+AR: we need to distinguish the general problem of type imports or structural types, and the problem of mutually recursive modules, they are separate problems to solve. The esoteric problem of mutually recursive modules is something we can impose limitations. In general, being able to link DAG-shaped dependency graphs in a reasonable fashion is something we want to support.
+
+TL: let’s take an AI to hash out goals around dynamic linking of modules. It has come up in issues that we cannot use nominal types to support these kind of examples. If we don’t care about these examples, it has a big impact on nominal type discussions.
+
+AR: nominal types is about linking, not at all specific for linking
+
+RT: but when we have those discussion, this problem is brought up to object nominal types
+
+AR: will give a presentation to explain the crucial points and why neither of the proposed solutions are solutions
+
+TL: will be great to hear those points. Will help to be crisp about what we are trying to achieve, the requirements. These examples come directly from conversations where you were arguing that nominal types don’t work. Surprised to hear you say these examples don’t matter.
+
+AR: yea there were many discussions, many confusions, and conflating various things. One of the big problem with the discussion we have, not focused, very tangential, not relevant to problems we are trying to solve.
+
+CW: One way to pare down is to fix a principle saying, we do not care about supporting separate compilation of mutually recursive type declarations. Then we can focus on what you consider to be more realistic objections of nominal types.
+
+AR: yea I can subscribe to those kind of statements.
+
+TL: Let's try to hammer out those principles by our next meeting, then we can be ready with a shared understanding of our goals.
+
+### Discussion: Deferred loading of types [Ross Tate]
+
+TODO(Ross Tate): insert link to slides
```

Comment:
@RossTate, please upload your slides when you get a chance!

Reply:
Apologies! I forgot this was on my to-do list. See #818